### PR TITLE
Issue 25545 when delete a page with experiments we must delete the experiments as well

### LINKED ISSUE #25545: When Delete a page with Experiments, we must Delete the Experiments as well
### Parent Issue

https://github.com/dotCMS/core/issues/22222

### Problem Statement

When we delete a Page with Experiments linked with it we are nor deleting the Experiments

### Steps to Reproduce

- Create a Page.
- Add an Experiment.
- Archived and Delete the Page
- Query directly the DataBase and you are going to realize that the Experiment is still there

### Acceptance Criteria

- We must delete all the Experiments linked with the Page when the Page is removed

### dotCMS Version

lastes in master

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by freddyucv**
### Proposed Changes
* When the Page is archived and you try to get the Experiment we were getting a error, this change fix it

https://github.com/dotCMS/core/pull/25794/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R759

*Delete the Page's Experiment when the Page is deleted

https://github.com/dotCMS/core/pull/25794/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R3106

* Not allow archive a Page when has a RUNNING Experiment

https://github.com/dotCMS/core/pull/25794/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R3568

* We need to delete the Experiment in this case no matter what is the Experiment's state so I create a new forceDelete method to delete without check the Experiment's Status

https://github.com/dotCMS/core/pull/25794/files#diff-5aa71567ff53ee229d25c2386cce3169c657f37a3e04cf27b75b3fd029537a7cR462

https://github.com/dotCMS/core/pull/25794/files#diff-5aa71567ff53ee229d25c2386cce3169c657f37a3e04cf27b75b3fd029537a7cR451

* When delete an Experiment we need delete the Experiment's Variant too

https://github.com/dotCMS/core/pull/25794/files#diff-5aa71567ff53ee229d25c2386cce3169c657f37a3e04cf27b75b3fd029537a7cR489

* When delete a Variant we need to delete the Contentlet's Version on tis Variant too 

https://github.com/dotCMS/core/pull/25794/files#diff-6b9c195cdec7e7bc927ed858f326518137ab896c29e1d05108f72898d006baf7R113

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 1]
```diff
@@ -3094,6 +3103,31 @@ private void deleteContentlet(final List<Contentlet> contentlets, final User use
...
+        }
+
+        final List<Experiment> pageExperiments = APILocator.getExperimentsAPI().list(
+                ExperimentFilter.builder().pageId(contentlet.getIdentifier()).build(), user);
+
+        for (final Experiment pageExperiment : pageExperiments) {
+            try {
+                APILocator.getExperimentsAPI().forceDelete(pageExperiment.id().orElseThrow(), user);
+            } catch (DotDataException | DotSecurityException e) {
+                final String message = String.format("Don;t be able to delete the experiment %s",
```
[the remark this answers is not in the document]
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:3124]: Actually it should be read as `Unable to delete experiment %s`

File: dotCMS/src/main/java/com/dotcms/variant/VariantAPIImpl.java [hunk 2]
```diff
@@ -105,6 +109,55 @@ public void delete(String id) throws DotDataException {
...
+    }
+
+    private static void deleteAllContentletVersions(Variant variant) throws DotDataException {
+        try {
+            APILocator.getContentletAPI().getAllContentByVariants(APILocator.systemUser(), false , variant.name())
+                    .stream()
+                    .map(contentlet -> getIdentifier(contentlet))
+                    .flatMap(identifier -> getVersions(variant, identifier).stream())
+                    .forEach(contentlet -> deleteContentlet(contentlet));
+        } catch (DotSecurityException e) {
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/variant/VariantAPIImpl.java:132]: How can we catch a `DotSecurityException` from whatever is inside the `.forEach()` consumer?

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/variant/VariantAPIImpl.java:132]: this method APILocator.getContentletAPI().getAllContentByVariants can throw a DotSecurityException outside of the forEach

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 3]
```diff
@@ -3516,13 +3550,30 @@ private void internalArchive(final Contentlet contentlet, final User user,
...
+    }
+
     private void internalArchive(final Contentlet contentlet, final User user,
             final boolean respectFrontendRoles,
             final boolean isDestroy, final IndexPolicy indexPolicy,
             final IndexPolicy indexPolicyDependencies,
             final Contentlet workingContentlet, final Contentlet liveContentlet)
             throws DotDataException, DotSecurityException {
 
+        final Optional<Experiment> experimentRunningOnPage = getExperimentRunningOnPage(contentlet);
```
[the remark this answers is not in the document]
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: done https://github.com/dotCMS/core/pull/25794/commits/a82e92a7833bf5ddcb2aa348a909baa5120a22c5#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82L3554

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: done https://github.com/dotCMS/core/pull/25794/commits/a82e92a7833bf5ddcb2aa348a909baa5120a22c5#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R3127

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**